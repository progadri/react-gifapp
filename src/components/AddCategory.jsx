import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {
  
    //? Aqui se define el valor por defecto que tendra el value del input;
    const [inputValue, setInputValue] = useState('');
  

    //? Esto sirve para capturar el valor que trae el input a través de la funcion onChange
    // const onInputChange = (e) => {
    //     console.log(e.target.value);
    //     setInputValue(e.target.value);
    // }

    //? aqui desestructuramos el event.target.value anterior para no tener que poner el event
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    };


    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories( [...categories,+InputValue ] );
        onNewCategory( inputValue.trim() );
        // setCategories( categories =>  [...categories,inputValue ] );
        setInputValue('');
    }
  
    return (
        <form onSubmit = {onSubmit} >
                <input 
                    type="text"
                    placeholder="Buscar gifs"
                    value={inputValue}
                    // onChange={ (e) => onInputChange(e)}
                    
                    //? Como el argumento que mandabamos a través de la función es el mismo se puede eliminar
                    onChange={onInputChange}
                />
        </form>

  )
}
