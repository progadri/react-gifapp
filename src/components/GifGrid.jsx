import PropTypes from 'prop-types';

// const api_key = '9xveTwzZOCootujPmn74B948KA68KUo6';

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({category}) => {

const {  images, isLoading } = useFetchGifs( category );

  return (
    <>
        <h3>{category}</h3>
        {
            isLoading && ( <h2>Cargando...</h2> )
            
        }
       
  
        <div className="card-grid">
            {
                images.map( ( image ) => (
                     <GifItem 
                     key={image.id}
                     //! el operador spreed pasa todas las properties
                     { ...image }

                     />
                ))
            }
           
        </div>
    </>
  )
};


GifGrid.propTypes = {
  category: PropTypes.string.isRequired,

}
