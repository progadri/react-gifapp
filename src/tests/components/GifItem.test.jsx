import {render, screen} from '@testing-library/react';
import { GifItem } from "../../components/GifItem"

describe('1.Pruebas en el componente GifItem', () => {

const title='Adrian';
const url= 'https://adrianrivera.es/'
    
test('1.1. Pruebas contra el snapshop', () => { 
    const {container} = render( <GifItem title={title} url={url}/> );
    expect ( container ).toMatchSnapshot();
 });

test('1.2. Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    
    render( <GifItem title={title} url={url}/> );
    // screen.debug();
    // expect ( screen.getByRole('img').src).toBe( url );
    // expect ( screen.getByRole('img').alt).toBe( title );
    const {src, alt} = screen.getByRole('img');
    expect ( src ).toBe( url );
    expect ( alt ).toBe( alt );

 });


 test('1.3. Debe de mostrar el titulo en el componente', () => { 
    
    render( <GifItem title={title} url={url}/> );

    expect(screen.getByText( title) ).toBeTruthy();


  })


})