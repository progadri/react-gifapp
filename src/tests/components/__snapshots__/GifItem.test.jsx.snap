// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.Pruebas en el componente GifItem 1.1. Pruebas contra el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Adrian"
      src="https://adrianrivera.es/"
    />
    <p>
      Adrian
    </p>
  </div>
</div>
`;
